bundles/ui/org.eclipse.smarthome.ui/src/main/java/org/eclipse/smarthome/ui/internal/proxy/ProxyServlet.java

```diff
@@ -147,96 +157,85 @@ public String getServletInfo() {
-
-                // copy all headers
-                while (iterator.hasNext()) {
-                    HttpField header = iterator.next();
-                    response.setHeader(header.getName(), header.getValue());
-                }
-
-            } catch (Exception e) {
-                if (e instanceof TimeoutException) {
-                    logger.warn("Proxy servlet failed to stream content due to a timeout");
-                    response.sendError(HttpServletResponse.SC_GATEWAY_TIMEOUT);
-                } else {
-                    logger.warn("Proxy servlet failed to stream content: {}", e.getMessage());
-                    response.sendError(HttpServletResponse.SC_BAD_REQUEST, e.getMessage());
-                }
-                return;
-            }
-            // now copy/stream the body content
-            try (InputStream responseContent = listener.getInputStream()) {
-                IOUtils.copy(responseContent, response.getOutputStream());
+                proxyRequest.header(HttpHeader.AUTHORIZATION, basicAuthentication);
             }
         }
     }
 
     /**
-     * Given a sitemap and widget, return the URI referenced by the widget in the sitemap.
-     * If the widget is not an Image or Video widget, then return <code>null</code>.
-     * If the widget is associated with an item, attempt to use the item's state as a URL.
-     * If the item's state as a string does not conform to RFC 2396, attempt to use the
-     * <code>url=</code> attribute in the sitemap. If that too does not conform to RFC 2396,
-     * then return <code>null</code>. In all cases where <code>null</code> is returned,
-     * this method first sends {@link HttpServletResponse.SC_BAD_REQUEST} back to the client.
+     * {@inheritDoc}
+     */
+    @Override
+    protected String rewriteTarget(HttpServletRequest request) {
+        return Objects.toString(uriFromRequest(request), null);
+    }
+
+    /**
+     * Determine which URI to address based on the request contents.
      *
-     * @param sitemapName
-     * @param widgetId
-     * @param response the HttpServletResponse to which detailed errors are sent
-     * @return the URI referenced by the widget
+     * @param request the servlet request
+     * @return the URI indicated by the request, or <code>null</code> if not possible
      */
-    private URI getURI(String sitemapName, String widgetId, HttpServletResponse response) throws IOException {
+    private URI uriFromRequest(HttpServletRequest request) {
 
+        // Return any URI we've already saved for this request
+        URI uri = (URI) request.getAttribute(ATTR_URI);
+        if (uri != null) {
+            return uri;
+        }
+
+        String sitemapName = request.getParameter("sitemap");
         if (sitemapName == null) {
-            response.sendError(HttpServletResponse.SC_BAD_REQUEST, "Parameter 'sitemap' must be provided!");
+            logger.error("Parameter 'sitemap' must be provided!");
```

Comment:
I do not really like such things to be logged as errors in the runtime log. After all, these might be unsolicited requests from remote clients and errors should rather turn up in the client (if on the server, it should rather be some http access/error log).
Is there no way to send an HTTP error back here as it was done before?